pkg/cassandra/version/version.go

```diff
@@ -0,0 +1,45 @@
+package version
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/coreos/go-semver/semver"
+)
+
+type Version struct {
+	*semver.Version
+}
+
+func New(v string) *Version {
+	return &Version{semver.New(v)}
+}
+
+func (v *Version) Equal(versionB *Version) bool {
+	semverA := *v.Version
+	semverB := *versionB.Version
+	return semverA.Equal(semverB)
+}
+
+// UnmarshalJSON handles Cassandra not-quite-semver versions
+// which may not have a patch version.
+// See https://github.com/apache/cassandra/blob/cassandra-3.11.2/src/java/org/apache/cassandra/utils/CassandraVersion.java
+func (v *Version) UnmarshalJSON(data []byte) error {
+	if v.Version == nil {
+		v.Version = &semver.Version{}
+	}
+	s, err := strconv.Unquote(string(data))
+	if err != nil {
+		return err
+	}
+	err1 := v.Version.Set(s)
+	if err1 != nil {
+		err2 := v.Version.Set(s + ".0")
+		if err2 != nil {
+			return err1
+		}
+	}
+	return nil
+}
+
+var _ json.Unmarshaler = &Version{}
```

Comment:
We also need to implement a marshaller that ensures that unmarshalling and then remarshalling is idempotent imo. This could cause confusion for users otherwise.

OTOH, perhaps we should actually disallow this invalid semver within Cassandra and *always* require valid semvers on our resources. If Cassandra reports an invalid semver, we could 'repair' it within the Pilot itself. Not sure of the implications of this, but given Cassandra is semver-like, it might work well & also saves our idempotentcy (I invented a new word) issues.

Reply:
I've switched to using Cassandra version throughout.
And added the MarshalJSON method.